homeassistant/components/tplink/diagnostics.py

```diff
@@ -36,6 +37,12 @@ async def async_get_config_entry_diagnostics(
 ) -> dict[str, Any]:
     """Return diagnostics for a config entry."""
     coordinator: TPLinkDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    internal_state = coordinator.device.internal_state
+    oui: str | None = None
+    if mac := internal_state.get("system", {}).get("get_sysinfo", {}).get("mac"):
```

Comment:
Instead of accessing the internal state, it'd be better to use the [`mac` property](https://github.com/python-kasa/python-kasa/blob/master/kasa/smartdevice.py#L451) as some devices store it under a different name.

This causes no separate I/O request when the sysinfo is already fetched.

Reply:
adjusted 👍